dotnet/src/Skills/Skills.OpenAPI/OpenApi/OpenApiDocumentParser.cs

```diff
@@ -31,7 +31,7 @@ public async Task<IList<RestApiOperation>> ParseAsync(Stream stream, Cancellatio
     {
         var jsonObject = await this.DowngradeDocumentVersionToSupportedOneAsync(stream, cancellationToken).ConfigureAwait(false);
 
-        using var memoryStream = new MemoryStream(Encoding.UTF8.GetBytes(jsonObject.ToJson()));
+        using var memoryStream = new MemoryStream(Encoding.UTF8.GetBytes(JsonSerializer.Serialize(jsonObject, SourceGenerationContext.WithGeneralOptions.JsonObject)));
```

Comment:
Serializing to a `string`, just to convert it to a Utf8 `byte[]` seems inefficient.

```suggestion
        using var memoryStream = new MemoryStream();
        JsonSerializer.Serialize(memoryStream, jsonObject, SourceGenerationContext.WithGeneralOptions.JsonObject);
        memoryStream.Position = 0;
```